Voting: Implement delegate voting with signatures (#27)
aragon/protocol: docs/4-entry-points/5-voting.md

- **State transitions:**
    - Save the controller address

### 4.5.2. Delegate

> - **Actor:** Any guardian that could potentially be drafted for an adjudication round or an external entity incentivized in a guardian of the Protocol

Reviewer: Similar to #26, I don't understand what "an external entity incentivized in a guardian of the Protocol" means.

Technically, any account is allowed to set a delegate here, so we could probably leave this as just "open".

- **Inputs:**
    - **Voter:** Address of the voter setting their delegate
    - **Delegate:** Address of the delegate to be set

> - **Authentication:** Open. Implicitly voters on their behalf or a whitelisted relayer.

Reviewer: ```suggestion
- **Authentication:** Open. The guardian or a whitelisted relayer.
```
